### PR TITLE
[Core]lm_connector with bytearray

### LINKED ISSUE #170: Code snippet about a more efficient lm_connector receiver
Hi,

As I said during our last meeting. It is better to pre-allocate memory to save time when receiving data from socket, I was trying to make a full implementation and issue a PR but I am kind of occupied recently. So I would like to leave an example prototype here FYI.

Hopefully I will be available soon but I don't want to block you guys' progress.

Below is my impl of receive all, it requires you to provide a preallocated buffer as input,
```
# assume buffer is allocated as a tensor using torch.zeros(length, dtype=torch.uint8, pin_memory=False)
# it would be better if we just reuse the page allocated by vLLM
def receive_all(self, n, buffer):
    bytes_received = 0
    ptr = buffer.data_ptr()  # Get the pointer to the start of the buffer

    while bytes_received < n:
        packet_size = self.client_socket.recv_into(
            (ctypes.c_char * (n - bytes_received)).from_address(ptr + bytes_received),
            n - bytes_received
        )
        if packet_size == 0:
            return None
        bytes_received += packet_size

    return buffer
```

### PR DESCRIPTION
**Pull request by Oasis-Git**
Modification on the receive_all api in lm_connector to avoid the reallocation during the buffer reading

FIX #170  (*link existing issues this PR will resolve*)

**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

<details>

<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to LMCache! Before submitting the pull request, please ensure the PR meets the following criteria. This helps us maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Please try to classify PRs for easy understanding of the type of changes. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Core]</code> for changes in the core LMCache logic (e.g., <code>LMCacheEngine</code>, <code>Backend</code> etc.)</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>Pass all linter checks. Please use <a href="https://github.com/LMCache/LMCache/blob/dev/format.sh"><code>format.sh</code></a> to format your code.</li>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li> Please include sufficient tests to ensure the change is stay correct and robust. This includes both unit tests and integration tests.</li>
</ul>

<h3>What to Expect for the Reviews</h3>

We aim to address all PRs in a timely manner. If no one reviews your PR within 5 days, please @-mention one of KuntaiDu, ApostaC or YaoJiayi.

### CODE REVIEW

**Comment by Oasis-Git** [comment]
A easy benchmark is done. For the local server the loading of buffer does not count much with regular length.
image

File: test.ipynb
Comment (ApostaC) [inline test.ipynb:1]: Can we remove this file? Thanks!

**Review by ApostaC — COMMENTED** [review]
LGTM and the benchmark graph looks pretty good! Just one quick comment:

File: test.ipynb
Comment (Oasis-Git) [reply test.ipynb:1]: Sorry for the late response. Yes, the unnecessary file is removed

**Review by YaoJiayi — APPROVED** [review]
LGTM!

**Pull request merged**